build-support/src/main/kotlin/bom.kt

```diff
@@ -0,0 +1,45 @@
+import org.gradle.api.Project
+import org.gradle.api.artifacts.dsl.DependencyConstraintHandler
+import org.gradle.kotlin.dsl.getByType
+import org.gradle.kotlin.dsl.withType
+import org.jetbrains.kotlin.gradle.dsl.KotlinJsProjectExtension
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.plugin.KotlinJsPluginWrapper
+import org.jetbrains.kotlin.gradle.plugin.KotlinMultiplatformPluginWrapper
+import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
+
+/**
+ * Collect all the root project's multiplatform targets and add them to the BOM.
+ *
+ * Only published subprojects are included.
+ *
+ * This supports Kotlin/Multiplatform and Kotlin/JS subprojects.
+ */
+fun Project.collectBomConstraints() {
```

Comment:
FYI @oldergod we might wanna copy-paste this into our other repos? Or maybe submit it as a new feature for the Vanniktech plugin? Note that I don’t yet support non-multiplatform Kotlin/JVM subprojects.

Reply:
Oooh, just realized I gotta support the Kotlin-Android plugin to capture the AssetFileSystem.

Reply:
I like it very much indeed

Reply:
Done.